### PR TITLE
Enhanced YAML Editor blog 

### PR DESCRIPTION
**Pull request by arunkumar611**
Need to record the video and add it.

### CODE REVIEW

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+meta_image: "meta.png"
+authors:
+  - pat-gavlin
+  - arun-loganathan
+tags:
+  - esc
+  - secrets
+---
+
+With [Pulumi ESC](/product/esc), our goal is to not only create a tool that simplifies the development process but also one that developers love. Building on this goal, we're excited to unveil the new Enhanced Pulumi ESC Editor, a big leap forward designed to improve the authoring experience for developers everywhere. This upgrade is focused on addressing the common challenges encountered during coding such as syntax errors and the need for frequent documentation lookups. Our aim is to make the process of authoring environments straightforward and remove the common hurdles for a smoother workflow.
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:16]:
```suggestion
With [Pulumi ESC](/product/esc), our goal is to not only create a tool that simplifies the development process but also one that developers love. In pursuit of this goal, we're excited to announce enhancements to the ESC environment editor. These enhancements are focused on addressing common challenges encountered when authoring environments: syntax errors, type errors, frequent context switches to and from documentation, and more. Our aim is to make the process of authoring environments as straightforward as possible by removing common hurdles.
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+<!--more-->
+
+## Highlights of the Enhanced Editor
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:20]:
```suggestion
## Highlights
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+- **Auto-complete**: This feature provides suggestions for code snippets and resource properties, eliminating the constant need to reference documentation. It's engineered to accelerate and refine the coding process, enabling developers to concentrate on innovation and problem-solving
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:22]:
```suggestion
- **Auto-complete**: This feature provides suggestions for code snippets and resource properties, easing the need to reference documentation. It is engineered to accelerate and refine the coding process, enabling developers to concentrate on innovation and problem-solving.
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- **Enhanced Diagnostics**: This goes beyond basic syntax checks by providing in-depth insights into potential errors, coupled with real-time fix recommendations. It serves as your personal coding assistant
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:23]:
```suggestion
- **Enhanced Diagnostics**: These diagnostics go beyond basic syntax checks by providing in-depth, as-you-type insights into errors.
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+- **Hover Documentation**: Simply hover over any piece of code to see a detailed tooltip with relevant insights and examples. This instant access to information reduces manual documentation lookups, streamlining the development process and minimizing disruptions
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:24]:
```suggestion
- **Hover Documentation**: Hover over any piece of code to see a detailed tooltip with relevant documentation and examples. This instant access to information reduces the need to context switch, streamlining the development process and minimizing disruptions.
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+Tap 'Ctrl+Space' on the Editor reveal the list of functions and properties.
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:26]:
```suggestion
Tap 'Ctrl+Space' in the editor reveal the list of functions and properties. 
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+## Highlights of the Enhanced Editor
+
+- **Auto-complete**: This feature provides suggestions for code snippets and resource properties, eliminating the constant need to reference documentation. It's engineered to accelerate and refine the coding process, enabling developers to concentrate on innovation and problem-solving
+- **Enhanced Diagnostics**: This goes beyond basic syntax checks by providing in-depth insights into potential errors, coupled with real-time fix recommendations. It serves as your personal coding assistant
+- **Hover Documentation**: Simply hover over any piece of code to see a detailed tooltip with relevant insights and examples. This instant access to information reduces manual documentation lookups, streamlining the development process and minimizing disruptions
+
+Tap 'Ctrl+Space' on the Editor reveal the list of functions and properties. 
+
+{{< video title="Pulumi ESC Enhanced Editor Demo" src="./esc-enhanced-editor-demo.mp4" autoplay="true" loop="true" >}}
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:28]:
```suggestion
{{< video title="Pulumi ESC editor enhancements demo" src="./esc-enhanced-editor-demo.mp4" autoplay="true" loop="true" >}}
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+## Contextual Information to your environments
+
+We also added support for getting rich contextual information such as current user, organization, environment name to the environments. You can use it to enrich audit logs, refine OIDC subject for granualar access and more! You can use the `context` attribute along with the following options to get contextual information:
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:32]:
```suggestion
We have also added support for accessing contextual information from within an ESC environment such as the current user, organization, and environment name. You can use this information to enrich audit logs, refine OIDC claims for more granular access, and more! This information is available by accessing the `context` property, and contains the following information:
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+* `context.rootEnvironment.name`: the name of the root environment being evaluated
+* `context.currentEnvironment.name`: the name of the current environment being evaluated
+* `context.user.login`: the user login identifier
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:36]:
```suggestion
* `context.user.login`: the login of the user evaluating the environment
```

File: themes/default/content/blog/esc-enhanced-editor/index.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+* `context.organization.login`: the organization login identifier
```
Comment (pgavlin) [inline themes/default/content/blog/esc-enhanced-editor/index.md:37]:
```suggestion
* `context.organization.login`: the name of the user's organization
```

**Review by pgavlin — CHANGES_REQUESTED** [review]
would prefer we not call this the "ESC Enhanced Editor", esp. because we will continue to make improvements here, and will probably want to blog about them each time.

also suggested a few edits to decrease wordiness.

**Comment by arunkumar611** [comment]
@pgavlin  - Thanks for the review. I updated the title and accepted your suggestions. Let me know if you have any other suggestions.

**Review by pgavlin — APPROVED** [review]

**Review by pgavlin — CHANGES_REQUESTED** [review]
oh we also need to change that title image

**Review by pgavlin — APPROVED** [review]

**Pull request merged**